crates/lib/src/store/mod.rs

```diff
@@ -169,9 +171,22 @@ impl BootedStorage {
                 }
                 let composefs = Arc::new(composefs);
 
+                // NOTE: This is assuming that we'll only have composefs in a UEFI system
+                // We do have this assumptions in a lot of other places
+                let parent = get_sysroot_parent_dev(&physical_root)?;
+                let (esp_part, ..) = get_esp_partition(&parent)?;
+                let esp_mount = mount_esp(&esp_part)?;
+
+                let boot_dir = match get_bootloader()? {
+                    Bootloader::Grub => physical_root.open_dir("boot").context("Opening boot")?,
+                    Bootloader::Systemd => esp_mount.fd.try_clone().context("Cloning fd")?,
```

Comment:
I don't think this will always be correct if XBOOTLDR is present; it's OK for now but can you add `// Note: Should also handle XBOOTLDR` or so?